src/components/CompareResults/ResultsTable.tsx

```diff
@@ -107,7 +107,26 @@ export default function ResultsTable() {
     new Map() as Map<string, Set<string>>, // ColumnID -> Set<Values to remove>
   );
 
+  useEffect(() => {
+    const filters = Array.from(rawSearchParams.entries())
+      .filter(([key]) => key.startsWith('filter'))
+      .reduce((accumulator: Map<string, Set<string>>, [key, value]) => {
+        const columnId = key.split('_')[1];
+        if (!accumulator.has(columnId)) {
+          accumulator.set(columnId, new Set());
+        }
+        const valuesArray = value.split(',').map((item) => item.trim());
+        valuesArray.forEach((item) => accumulator.get(columnId)?.add(item));
+        return accumulator;
+      }, new Map<string, Set<string>>());
+
+    setTableFilters(filters);
+  }, [rawSearchParams]);
+
```

Comment:
This is a good effort! 

Now, I didn't mention this in the last PRs trying to solve this issue, but the problem can also be solved _without_ a `useEffect` (esp to prevent unnecessary rerenders). 

You could store all the logic for setting table filters in a separate, custom hook, ex: `useTableFilters` by putting it in the hooks folder; ex: `src/hooks/useTableFilters` 

You could also utilize [`useMemo`](https://react.dev/reference/react/useMemo) to compute the table filters; `useMemo` will make sure your calculations only run when `rawSearchParams` changes.

And then you can import the custom hook in `ResultsTable` and execute it in place of your `useEffect`:

 `const tableFilters = useTableFilters();`

What do you think?

Reply:
Thank you for your review.

I agree that moving the logic into a custom hook and using `useMemo` to memorize the filters make sense, as it ensures that the filters are only recalculated when `rawSearchParams` change, potentially improving performance.

Since the custom hook will handle setting the `tableFilters` when the filter parameters change, I think we can remove `setTableFilters` from the `onClearFilter` and `onToggleFilter` functions, solely focusing on updating the URL parameters instead. 

We could then move `onClearFilter` and `onToggleFilter` functions into the custom hook together with setting table filters logic which will make it easier to reuse the hook in other components if needed.

We can import the custom hook in `ResultsTable` as 
`const { tableFilters, onClearFilter, onToggleFilter } = useTableFilters()`

Please correct me if I am wrong. 

Please also let me know if you would like me to implement this. I am happy to do so.

Reply:
That sounds like a great idea! Please go ahead and try it.

Reply:
Thanks. I am working on it.

Reply:
Hi there,

I have created a custom hook to manage the table filter logic. 

However, I noticed that removing the `setTableFilters' from `onClearFilter` and `onToggleFilter` functions, leaving them only with updating the URL parameters, resulted in the UI not reflecting to the new `tableFilters`. 

I believe this is because  React does not recognize this change as a state update in the component until the page is refreshed/re-render, which is necessary for React to detect the updated URL parameters.

For now, I keep the `setTableFilters` in these functions, as we need a state update for React to dynamically reflect the changes made. 

Please let me know if I am missing something in my thinking. Thank you.

Reply:
Yes, keep the `setTableFilters` in these functions. I missed the part where you suggested removing it from `onToggleFilter` and `onClearFilter`